src/headers/tomcrypt_cfg.h

```diff
@@ -277,6 +277,15 @@ typedef unsigned long ltc_mp_digit;
    #define LTC_HAVE_BSWAP_BUILTIN
 #endif
 
+#ifdef __GNUC__
+   #define LTC_DEPRECATED __attribute__((deprecated))
+#elif defined(_MSC_VER)
+   #define LTC_DEPRECATED __declspec(deprecated)
+#endif
+
+#ifndef LTC_DEPRECATED
+   #error "You need to define LTC_DEPRECATED for this compiler"
```

Comment:
I do not like this. It is very unfriendly to non-gcc/non-msvc compilers.

Reply:
I hoped you're going to build this on "the other compilers" and then either fill it in or define it to nothing if the "deprecation" feature doesn't exist for "the other compilers"

Reply:
Consider:
```C
#ifdef __GNUC__
   #define LTC_DEPRECATED __attribute__((deprecated))
#elif defined(_MSC_VER)
   #define LTC_DEPRECATED __declspec(deprecated)
#else
   #define LTC_DEPRECATED
#endif

It might also cause troubles on older gcc as I do not know since what version this attribute is supported.
```

Reply:
looks like [gcc-3.1](https://gcc.gnu.org/onlinedocs/gcc-3.1/gcc/Type-Attributes.html) introduced it as it wasn't in [gcc-3.0.4](https://gcc.gnu.org/onlinedocs/gcc-3.0.4/gcc/Type-Attributes.html)

Reply:
It seems to be supported on gcc 3.1+

Reply:
`#if (__GNUC__ > 3) || (__GNUC__ == 3 && __GNUC_MINOR__ >= 0)`